sdk/core/Azure.Core/src/Http/ConditionalRequestOptions.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Core.Http
+{
+    /// <summary>
+    /// Specifies HTTP options for conditional requests.
+    /// </summary>
+    public class ConditionalRequestOptions : IEquatable<ConditionalRequestOptions>
+    {
+        private const string IfMatchHeader = "If-Match";
+        private const string IfNoneMatchHeader = "If-None-Match";
+
+        /// <summary>
+        /// Optionally limit requests to resources that have a matching ETag.
+        /// </summary>
+        public ETag? IfMatch { get; set; }
+
+        /// <summary>
+        /// Optionally limit requests to resources that do not match the ETag.
+        /// </summary>
+        public ETag? IfNoneMatch { get; set; }
+
+        /// <summary>
+        /// Set preconditions that indicate to apply an operation only if the
+        /// target resource has changed.
+        /// </summary>
+        /// <param name="etag">The version of the resource to compare to.</param>
+        public virtual void SetIfModifiedCondition(ETag etag)
```

Comment:
Does it make sense for users to call multiple of these `SetXyzConditions` methods on the same `ConditionalRequestOptions`?  I thought of them more as static helpers to create a `ConditionalRequestOptions` using a friendlier high-level name.

Reply:
I think you can combine match and modified since in theory. 


> static helpers to create a ConditionalRequestOptions

Doesn't work when inheriting from ConditionalRequestOptions, statics would return base class instead of derived one.